packages/forms-web-app/src/middleware/check-logged-in.js

```diff
@@ -0,0 +1,46 @@
+const { getAppealUserSession } = require('../services/appeal-user.service');
+const { isFeatureActive } = require('../featureFlag');
+const { FLAG } = require('@pins/common/src/feature-flags');
+const {
+	constants: { NEW_OR_SAVED_APPEAL_OPTION }
+} = require('@pins/business-rules');
+
+/**
+ * @type {import('express').RequestHandler}
+ */
+const isLoggedIn = async (req, res, next) => {
+	const enrolUsers = await isFeatureActive(FLAG.ENROL_USERS);
+
+	if (!enrolUsers) {
+		next();
+	}
+
+	const user = getAppealUserSession(req);
+
+	if (user && user?.id) {
```

Comment:
<img width="556" alt="Screenshot 2024-01-10 at 11 01 36" src="https://github.com/Planning-Inspectorate/appeal-planning-decision/assets/35689158/b87e6e20-0f66-455b-94a1-3d1eb41398d6">

no need for optional chaining here, the jit just moves on if the first condition doesn't pass